ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,309 @@
+        metadata["image_ids"] = [(image_id[0].name, image_id[1]) for image_id in self.image_ids]
+        metadata["filters"] = self.filter_metadata
+        with open(metadata_filepath, 'w') as outfile:
+            json.dump(metadata, outfile) 
+
+    def write_dataset(self, associated_files):
+        """Method is parent function for writing out complete dataset. Method first
+            creates 'test' and 'dev' subsets. The 'test' subset gets all related files
+            to it copied into a test folder. The 'dev' subset calls 'write_out_complete_set'
+            in the 'splits/complete' directory. Note that prior to this method, obj_list
+            should be set to a list of objects that are meant to be written.
+
+            If overloaded, there are no expectations, but note that the variables 'kfolds'
+            and 'test_percent' are provided for use.
+        
+        Args:
+            associated_files (list): decides what files are to be copied for the test set
+
+        Variables Needed:
+            obj_list (list): list of objects to be written in dataset
+            dataset_path (Path): where dataset will be written (provided by 'create' input)
+            dataset_name (str): the name of the dataset (provided by 'create' input)
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        print(dataset_path)
+
+        test_subset, dev_subset = split_data(self.obj_list, test_percent=self.test_percent)
+
+        # Test subset
+        test_path = dataset_path / 'test'
+        self.write_out_test_set(test_path, test_subset, associated_files)
+
+        dev_path = dataset_path / 'splits'
+
+        # standard_path = dev_path / 'standard'
+        # write_out_fold(standard_path, fold, is_standard=True)
+
+        complete_path = dev_path / 'complete'
+        self.write_out_complete_set(complete_path, dev_subset)
+
+    def write_out_test_set(self, path, data, associated_files):
+        """Method is helper function for writing out dataset. Writes
+            out test set by copying over associated files to the
+            specified test path. Assumes objlist has 'image_filepath'
+            and 'image_id' as keys.
+
+            If overloaded, there are no expectations.
+
+        Args:
+            path (Path): Path to where data should be written
+            data (list): data that should be written
+            associated_files (list): decides what files are to be copied for the test set
+        """
+        os.mkdir(path)
+        test_image_ids = [(Path(os.path.dirname(obj['image_filepath'])), obj['image_id']) for obj in data]
+        copy_associated_files(test_image_ids, path, associated_files)
+
+    def write_out_complete_set(self, path, data):
+        """Method is helper function for writing out dataset. Creates a 
+            'train' subdirectory and calls for 'write_data' to write
+            test_data and _train_data.
+
+            If overloaded, there are no expectations.
```

Comment:
This is a nitpick, because people will generally know what you mean, but all instances of "overload" should technically say "override"